### DISCUSSION TITLE
Can't generate SSL certificate

### DISCUSSION THREAD

**Discussion by mistralnewman** [mastodon/mastodon#17372] [Q&A] [answered]
Hello all! I'm currently trying to set up a Mastodon instance, but I'm a bit stuck on the step of generating an SSL certificate. When I run `certbot --nginx -d example.com` (with my website's URL), I get this error:

```
Error while running nginx -c /etc/nginx/nginx.conf -t.

nginx: [emerg] no "ssl_certificate" is defined for the "listen ... ssl" directive in /etc/nginx/sites-enabled/mastodon:25
nginx: configuration file /etc/nginx/nginx.conf test failed
```

I've tried uncommenting the lines with SSL certificate information in the config file, but then I get a different error saying it cannot load the SSL certificate, which makes sense. I've also tried commenting out the entire port 443 block, but then I get this error:

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/certbot/error_handler.py", line 124, in _call_registered
    self.funcs[-1]()
  File "/usr/lib/python3/dist-packages/certbot/auth_handler.py", line 243, in _cleanup_challenges
    self.auth.cleanup(achalls)
  File "/usr/lib/python3/dist-packages/certbot_nginx/configurator.py", line 1152, in cleanup
    self.restart()
  File "/usr/lib/python3/dist-packages/certbot_nginx/configurator.py", line 919, in restart
    nginx_restart(self.conf('ctl'), self.nginx_conf)
  File "/usr/lib/python3/dist-packages/certbot_nginx/configurator.py", line 1202, in nginx_restart
    raise errors.MisconfigurationError(
certbot.errors.MisconfigurationError: nginx restart failed:
b''
b''
```
It seems like I can't start nginx without having a certificate generated, but I can't generate a certificate without starting nginx. I found [a forum post with a similar problem](https://discourse.joinmastodon.org/t/nginx-reverse-proxy-on-another-server/485/7), but they were using a reverse proxy so I don't think it applies to my situation. Does anyone have any recommendations for what I can do here? Happy to share additional information if necessary. Thanks!

**Comment by Fireshaded** [comment, marked as answer]
I ran into the same answer and it seems it's a chicken and egg problem where Nginx needs the certs to run, while certbot needs Nginx to pass the text that the config is valid.

Try using stopping Nginx `sudo systemctl stop nginx`, if using Ubuntu/Debian.

Then generate the cert using `sudo certbot certonly --standalone -d example.com`, this worked for me after bashing my head against the wall with the same problem.

**Reply by mistralnewman** [reply]
This worked, thanks so much!

**Reply by Fireshaded** [reply]
No worries!

Be curious if you run into another problem once completing the guide that the server runs but you don't get the log on screen and only the image of the elephant with the computer saying error. My current problem at the moment.

**Reply by mistralnewman** [reply]
I seem to be having a different problem where the site is up and running with a Mastodon favicon but there is no HTML being served and all pages are totally blank. I'm going to try some things before asking for more help, though, since I've only had this problem for about ten minutes. Best of luck with your server!

**Comment by i0ivi0i** [comment]
The problem I'm having is:
Error while running nginx -c /etc/nginx/nginx.conf -t.

nginx: [emerg] could not build server_names_hash, you should increase server_names_hash_bucket_size: 64
nginx: configuration file /etc/nginx/nginx.conf test failed

The nginx plugin is not working; there may be problems with your existing configuration.
The error was: MisconfigurationError('Error while running nginx -c /etc/nginx/nginx.conf -t.\n\nnginx: [emerg] could not build server_names_hash, you should increase server_names_hash_bucket_size: 64\nnginx: configuration file /etc/nginx/nginx.conf test failed\n')


what do I do
